storage/store/my_receipts.go

```diff
@@ -54,19 +106,9 @@ func (m *MyExecutionReceipts) myReceipt(blockID flow.Identifier) (*flow.Executio
 // No errors are expected during normal operation
 // If entity fails marshalling, the error is wrapped in a generic error and returned.
 // If Badger unexpectedly fails to process the request, the error is wrapped in a generic error and returned.
+// If a different my receipt has been indexed for the same block, the error is wrapped in a generic error and returned.
 func (m *MyExecutionReceipts) BatchStoreMyReceipt(receipt *flow.ExecutionReceipt, rw storage.ReaderBatchWriter) error {
-
-	err := m.genericReceipts.BatchStore(receipt, rw)
-	if err != nil {
-		return fmt.Errorf("cannot batch store generic execution receipt inside my execution receipt batch store: %w", err)
-	}
-
-	err = operation.IndexOwnExecutionReceipt(rw.Writer(), receipt.ExecutionResult.BlockID, receipt.ID())
-	if err != nil {
-		return fmt.Errorf("cannot batch index own execution receipt inside my execution receipt batch store: %w", err)
-	}
-
-	return nil
+	return m.cache.PutTx(rw, receipt.ExecutionResult.BlockID, receipt)
```

Comment:
This is a bit nit-picky but the `cache` is not really just a cache which is very apparent here. If `cache` was just a cache, `put` would not be sufficient. The `Cache` object is really a cache wrapped storage. Maybe we need to rename the `Cache` to `CachedStorage` or something.

Reply:
Yeah, agree. Will rename in a separate PR altogether.